### PR TITLE
Fix padding length for sglang rollout in veRL

### PR DESCRIPTION
**Pull request by GeLee-Q**
Fixed a portion of the issues encountered during VLM GPTO training as mentioned in the article. 
https://github.com/zhaochenyang20/Awesome-ML-SYS-Tutorial/blob/main/rlhf/verl/veRL-VLM.md

When do_sample=False, different models under DP output sequences of inconsistent lengths, which may be padded to different lengths, ultimately causing shape inconsistencies during output and resulting in errors in collect_dp_compute_data_proto. The following situation occurred: 
```
DataProto(batch=TensorDict(
    fields={
        attention_mask: Tensor(shape=torch.Size([151, 5120]), device=cpu, dtype=torch.int64, is_shared=False),
        input_ids: Tensor(shape=torch.Size([151, 5120]), device=cpu, dtype=torch.int64, is_shared=False),
        position_ids: Tensor(shape=torch.Size([151, 5120]), device=cpu, dtype=torch.int64, is_shared=False),
        prompts: Tensor(shape=torch.Size([151, 1024]), device=cpu, dtype=torch.int64, is_shared=False),
        responses: Tensor(shape=torch.Size([151, 4096]), device=cpu, dtype=torch.int64, is_shared=False)},
    batch_size=torch.Size([151]),
    device=cpu,
    is_shared=False), non_tensor_batch={}, meta_info={}), DataProto(batch=TensorDict(
    fields={
        attention_mask: Tensor(shape=torch.Size([151, 5120]), device=cpu, dtype=torch.int64, is_shared=False),
        input_ids: Tensor(shape=torch.Size([151, 5120]), device=cpu, dtype=torch.int64, is_shared=False),
        position_ids: Tensor(shape=torch.Size([151, 5120]), device=cpu, dtype=torch.int64, is_shared=False),
        prompts: Tensor(shape=torch.Size([151, 1024]), device=cpu, dtype=torch.int64, is_shared=False),
        responses: Tensor(shape=torch.Size([151, 4096]), device=cpu, dtype=torch.int64, is_shared=False)},
    batch_size=torch.Size([151]),
    device=cpu,
    is_shared=False), non_tensor_batch={}, meta_info={}), DataProto(batch=TensorDict(
    fields={
        attention_mask: Tensor(shape=torch.Size([151, 3072]), device=cpu, dtype=torch.int64, is_shared=False),
        input_ids: Tensor(shape=torch.Size([151, 3072]), device=cpu, dtype=torch.int64, is_shared=False),
        position_ids: Tensor(shape=torch.Size([151, 3072]), device=cpu, dtype=torch.int64, is_shared=False),
        prompts: Tensor(shape=torch.Size([151, 1024]), device=cpu, dtype=torch.int64, is_shared=False),
        responses: Tensor(shape=torch.Size([151, 2048]), device=cpu, dtype=torch.int64, is_shared=False)},
    batch_size=torch.Size([151]),
    device=cpu,
    is_shared=False), non_tensor_batch={}, meta_info={}), DataProto(batch=TensorDict(
    fields={
        attention_mask: Tensor(shape=torch.Size([151, 3072]), device=cpu, dtype=torch.int64, is_shared=False),
        input_ids: Tensor(shape=torch.Size([151, 3072]), device=cpu, dtype=torch.int64, is_shared=False),
        position_ids: Tensor(shape=torch.Size([151, 3072]), device=cpu, dtype=torch.int64, is_shared=False),
        prompts: Tensor(shape=torch.Size([151, 1024]), device=cpu, dtype=torch.int64, is_shared=False),
        responses: Tensor(shape=torch.Size([151, 2048]), device=cpu, dtype=torch.int64, is_shared=False)},
    batch_size=torch.Size([151]),
    device=cpu,
    is_shared=False), non_tensor_batch={}, meta_info={})]
```

This modification resolves this issue.

### CODE REVIEW

File: verl/workers/rollout/sglang_rollout/sglang_rollout.py [hunk 1]
```diff
@@ -232,6 +233,7 @@ def generate_sequences(self, prompts: DataProto, **kwargs) -> DataProto:
                 skip_special_tokens=True,
                 spaces_between_special_tokens=True,
             )
+            pad_length = max(pad_length, 4096)
```
Comment (vermouth1992) [inline verl/workers/rollout/sglang_rollout/sglang_rollout.py]: Why max with 4096?

Comment (GeLee-Q) [reply verl/workers/rollout/sglang_rollout/sglang_rollout.py]: the kwargs max_new_tokens is 4096
```
kwargs = dict(
                n=1,
                presence_penalty=0.0,
                frequency_penalty=0.0,
                repetition_penalty=1.0,
                temperature=0,
                top_p=1,
                top_k=-1,
                ignore_eos=False,
                min_new_tokens=0,
                max_new_tokens=4096,
                skip_special_tokens=True,
                spaces_between_special_tokens=True,
            )
```

Comment (PeterSH6) [reply verl/workers/rollout/sglang_rollout/sglang_rollout.py]: Why is the minimum pad length set to 4096?
It would add too much unnecessary padding in some case.

Comment (GeLee-Q) [reply verl/workers/rollout/sglang_rollout/sglang_rollout.py]: The reason it's 4096 is to align the parameters in kwargs. This code will only run when do_sample is false, which is during the validation stage. 
```
        # perform validation before training
        # currently, we only support validation using the reward_function.
        if self.val_reward_fn is not None and self.config.trainer.get('val_before_train', True):
            val_metrics = self._validate()
            pprint(f'Initial validation metrics: {val_metrics}')
            logger.log(data=val_metrics, step=self.global_steps)
            if self.config.trainer.get('val_only', False):
                return
```

During the actual rollout stage, the pad value will be self.config.response_length.

**Comment by zhaochenyang20** [comment]
I asked yitong to review. thanks!

**Comment by GeLee-Q** [comment]
@ocss884 After communicating with Junrong  , max_new_tokens=4096 is not a special parameter, but instead causes torch.cat errors and OOM issues during the rollout process. Therefore, changed to max_new_tokens=self.config.response_length

**Review by ocss884 — APPROVED** [review]
LGTM!

**Review by vermouth1992 — APPROVED** [review]

**Comment by zhaochenyang20** [comment]
gteat!
